github-gist.c

```diff
@@ -0,0 +1,53 @@
+#define _GNU_SOURCE /* asprintf() */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+#include "cee-utils.h"
+#include "cee-utils/ntl.h"
+#include "json-actor.h"
+
+#include "github.h"
+#include "github-internal.h"
+
+ORCAcode
+github_create_gist(struct github *client, struct github_gist_create_params *params, struct github_gist *gist)
+{
+  log_info("===create-gist===");
+
+  if (!params->description) {
+    log_error("Missing 'description'");
+  }
+  if (!params->title) {
+    log_error("Missing 'title'");
+  }
+  if (!params->contents) {
+    log_error("Missing 'contents'");
+  }
+
+  char payload[4096];
+  char fmt[2048];
+
+  /* Create the format string for the payload
+   * TODO:
+   * Allocate buffer big enough, then free it after the request is made
+   * */
+  snprintf(fmt, sizeof(fmt), "(public): \"%s\", (description): \"%s\", (files): { (%s): { (content): \"%s\" }}", params->public,
+                                                                                                                 params->description,
+                                                                                                                 params->title,
+                                                                                                                 params->contents);
+  printf("%s\n", fmt);
```

Comment:
Is this printf() a debug remnant, or is it here for logging purposes?

Reply:
It was a debug remnant, it has been removed.